src/common/snippets/src/lowered/pass/insert_buffers.cpp

```diff
@@ -133,6 +136,10 @@ LinearIR::constExprIt InsertBuffers::insertion_position(const LinearIR& linear_i
         const auto down_loop_id = down_loops[loop_idx];
         return loop_manager->get_loop_bounds(linear_ir, down_loop_id).first;
     }
+    // If upper and lower expressions are in the same loop, we should insert Buffer between them
+    if (loop_idx == up_loop_count && loop_idx == down_loop_count) {
```

Comment:
Good catch! We didn't have this case before.
Maybe it'd be better if we insert this check in L118?
Smth like
`if (down_loops.empty() || down_loops == upper_loops)`
What do you think?

Reply:
Why would it be better?
The worst-case complexity of `down_loops == up_loops` is `O(N)`, where N is the vector length.
The worst case complexity of the present solution is `O(1)`.

Also, I think that it is more readable, since you cover all meaningful `loop_idx` values/ranges in the same place. However, readability is a perceptional thing, so I guess your motivation was to improve it as well 🙂